tcmur_cmd_handler.c

```diff
@@ -538,8 +540,11 @@ static int handle_unmap(struct tcmu_device *dev, struct tcmulib_cmd *origcmd)
 	}
 
 	state = unmap_state_alloc(dev, origcmd, &ret);
-	if (!state)
+	if (!state) {
+		tcmu_dev_dbg(dev,"unmap no resources to allocate struct\n");
+		ret = TCMU_STS_NO_RESOURCE;
```

Comment:
This should be set by unmap_state_alloc on failure. If not then that function needs to be fixed.